plugins/modules/ipbase_facts.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Dominik Kukacka <dominik.kukacka@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: ipbase_facts
+short_description: Retrieve IP geolocation and other facts of a host's IP address
+description:
+  - "Retrieve IP geolocation and other facts of a host's IP address using the ipbase.com API"
+author: "Dominik Kukacka (@dominikkukacka)"
+extends_documentation_fragment:
+  - community.general.attributes
+  - community.general.attributes.facts
+  - community.general.attributes.facts_module
+options:
+  http_agent:
+    description:
+      - Set http user agent
+    required: false
+    default: "ansible-ipbase-module/0.0.1"
+    type: str
+notes:
+  - "Check http://ipbase.com/ for more information"
+'''
+
+EXAMPLES = '''
+# Retrieve IP geolocation and other facts of a host's IP address
+- name: Get IP geolocation information
+  community.general.ipbase_facts:
+'''
+
+RETURN = '''
+ansible_facts:
+  description: "Dictionary of ip facts for a host's IP address"
+  returned: changed
+  type: complex
+  contains:
+    ip:
+      description: "Public IP address of the host"
```

Comment:
Always put full sentences in paragraphs.
```suggestion
      description: "Public IP address of the host."
```